pkg/kubelet/certificate/kubelet.go

```diff
@@ -94,8 +94,13 @@ func NewKubeletServerCertificateManager(kubeClient clientset.Interface, kubeCfg
 
 	getTemplate := func() *x509.CertificateRequest {
 		hostnames, ips := addressesToHostnamesAndIPs(getAddresses())
-		// don't return a template if we have no addresses to request for
-		if len(hostnames) == 0 && len(ips) == 0 {
+		// don't return a template if we have no addresses to request for.
+		// A Kubernetes Node requires to have at minimum one IP address
+		// because those are used on the Pods field HostIPs and in some cases,
```

Comment:
it uses the IP from node.status.addresses as pod hostIP?

Reply:
yes, see https://docs.google.com/document/d/1mqdVLQHIYjrzjy8Hq-FMysthHRL7ht0guk2uWI0FxKM/edit#heading=h.97ewrhxwu05m fore more detailed explanation

Reply:
api docs https://github.com/kubernetes/kubernetes/blob/2ad04a0505ef13915e11d8b2b3445e514065a9bb/staging/src/k8s.io/api/core/v1/types.go#L6148-L6164